tests/unit/common/test_table.py

```diff
@@ -336,6 +319,128 @@ def setUpContainer(self):
         return table
 
 
+class TestDynamicTableRegion(TestCase):
+
+    def setUp(self):
+        self.spec = [
+            {'name': 'foo', 'description': 'foo column'},
+            {'name': 'bar', 'description': 'bar column'},
+            {'name': 'baz', 'description': 'baz column'},
+        ]
+        self.data = [
+            [1, 2, 3, 4, 5],
+            [10.0, 20.0, 30.0, 40.0, 50.0],
+            ['cat', 'dog', 'bird', 'fish', 'lizard']
+        ]
+
+    def with_columns_and_data(self):
+        columns = [
+            VectorData(name=s['name'], description=s['description'], data=d)
+            for s, d in zip(self.spec, self.data)
+        ]
+        return DynamicTable("with_columns_and_data", 'a test table', columns=columns)
+
+    def test_indexed_dynamic_table_region(self):
+        table = self.with_columns_and_data()
+        dynamic_table_region = DynamicTableRegion('dtr', [1, 2, 2], 'desc', table=table)
+        fetch_ids = dynamic_table_region[:3].index.values
+        self.assertListEqual(fetch_ids.tolist(), [1, 2, 2])
+
+    def test_dynamic_table_region_iteration(self):
+        table = self.with_columns_and_data()
+        dynamic_table_region = DynamicTableRegion('dtr', [0, 1, 2, 3, 4], 'desc', table=table)
+        for ii, item in enumerate(dynamic_table_region):
+            self.assertTrue(table[ii].equals(item))
+
+    def test_dynamic_table_region_shape(self):
+        table = self.with_columns_and_data()
+        dynamic_table_region = DynamicTableRegion('dtr', [0, 1, 2, 3, 4], 'desc', table=table)
+        self.assertTupleEqual(dynamic_table_region.shape, (5, 3))
+
+    def test_dynamic_table_region_to_dataframe(self):
+        table = self.with_columns_and_data()
+        dynamic_table_region = DynamicTableRegion('dtr', [0, 1, 2, 2], 'desc', table=table)
+        res = dynamic_table_region.to_dataframe()
+        self.assertListEqual(res.index.to_list(), [0, 1, 2, 2])
```

Comment:
Pandas 0.24.0 introduced the alias `to_list` for `tolist`. For compatibility with earlier versions of pandas, let's use `tolist` instead of `to_list`. 

https://stackoverflow.com/questions/57854228/pandas-dataframe-index-to-list-vs-tolist

Reply:
We currently list pandas 0.23 as a minimum requirement.

Reply:
Sounds good. I just updated the tests accordingly to use ``tolist`` instead of ``to_list``.